app/javascript/components/public_health/query/AdvancedFilter.js

```diff
@@ -434,8 +434,8 @@ class AdvancedFilter extends React.Component {
     axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
     axios
       .post(window.BASE_PATH + '/user_filters', { activeFilterOptions: this.state.activeFilterOptions, name: this.state.filterName })
-      .catch(() => {
-        toast.error('Failed to save filter.');
+      .catch(err => {
+        toast.error(err?.response?.data?.error ? err.response.data.error : 'Failed to update filter.');
```

Comment:
Under what circumstances would there be an error updating the filter? Would a filter not existing qualify? If I'm understanding this correctly, this call hits the `UserFiltersController#update` method?

If I...
- open the dashboard in two tabs
- delete my filter in one tab
- try to update the same filter in the other 
then the page doesn't present any message that something is wrong. I think that's due to the lone return conditional on L46 of the `UserFiltersController`. It appears that the update method will currently only trigger this error message if an exception occurs.

Reply:
In the circumstance addressed in the changes above, where the user hits the max number of allowed filters. This is a `POST` to the `UserFiltersController#create` method.

Reply:
If we are creating, then should the default error be `Failed to create filter` instead of `failed to update filter`?

Reply:
Yep, good catch - changed to "Failed to save filter" here: 4ab7044